### PR TITLE
Try to add documentation in socket.hpp

### PR DESCRIPTION
**Pull request by tkhurana96**
But at "!!Improve This!!" sentences need to be completed.

### CODE REVIEW

File: include/socket.hpp [hunk 1]
```diff
@@ -21,7 +26,16 @@ class Socket final {
...
 
-
+	/** @method low_write
+	* @access private
+	* Writes given _msg using _sockfd by calling _fn with args having flags and destination socket address.
+	* 
+	* @param {callable} _fn Some callable that writes using socket descriptor.
+	* @param {int} _sockfd Descriptor representing a socket.
+	* @param {string} _msg Msg to write on _sockfd.
+	* @param {_______} args Flags, destination sockaddr objects and their lengths. 		!!Improve This!!
```
Comment (agauniyal) [inline include/socket.hpp]: _____ = parameter_pack

File: include/socket.hpp [hunk 2]
```diff
@@ -37,7 +51,16 @@ class Socket final {
...
 
-
+	/** @method low_read
+	* @access private
+	* Reads using _sockfd by calling _fn  with args having flags and destination socket address.
+	* 
+	* @param {callable} _fn Some callable that reads using socket descriptor.
+	* @param {int} _sockfd Descriptor representing a socket.
+	* @param {string} _str String to store the data.
+	* @param {_______} args Flags, destination sockaddr objects and their lengths. 		!!Improve This!!
```
Comment (agauniyal) [inline include/socket.hpp]: _____ = parameter_pack

File: include/socket.hpp [hunk 3]
```diff
@@ -9,6 +9,11 @@
 
 namespace net {
```
Comment (agauniyal) [inline include/socket.hpp:10]: Please format with `clang-format`.

File: include/socket.hpp [hunk 4]
```diff
@@ -181,8 +244,26 @@ class Socket final {
...
+	* 
+	* @param {char []} _addr Ip address in case of ipv4 or ipv6 domain, and Path in case of unix domain.
+	* @param {int} _port Port number in case of addrIpv4 or addrIpv6
+	* @param {bool *} _errorNB To signal error in case of non-blocking connect.
+	*/
 	void connect(const char[], const int = 0, bool * = nullptr);
 
+	/** @method connect
+	* @access public
+	* Connects net::Socket to _______ if successful else throws runtime_error exception.		!!Improve This!!
```
Comment (agauniyal) [inline include/socket.hpp]: _______ = ipv4 peer

File: include/socket.hpp [hunk 5]
```diff
@@ -211,7 +292,16 @@ class Socket final {
 		}
 	}
 
-
+	/** @method connect
+	* @access public
+	* Connects net::Socket to _______ if successful else throws runtime_error exception.		!!Improve This!!
```
Comment (agauniyal) [inline include/socket.hpp]: _______ = ipv6 peer

File: include/socket.hpp [hunk 6]
```diff
@@ -240,7 +330,16 @@ class Socket final {
...
```
Comment (agauniyal) [inline include/socket.hpp]: _______ = unix socket peer

File: include/socket.hpp [hunk 7]
```diff
@@ -493,14 +719,38 @@ class Socket final {
...
+	*/
 	SF::sockOpt getOpt(SF::opt) const;
 
+	/** @method stop
+	* @access public
+	* Shutdown Socket using net::SF::shut.
+	* 
+	* @param {net::SF::shut} s Option specifying which side of connection to shutdown for Socket.
+	*/
 	void stop(SF::shut s) const noexcept
```
Comment (agauniyal) [inline include/socket.hpp]: Please make it `_s` instead of `s` both in documentation and function definition.

File: include/socket.hpp [hunk 8]
```diff
@@ -493,14 +719,38 @@ class Socket final {
 	// recv method calls
 
 
+	/** @method setOpt
+	* @access public
+	* Set a socket option from net::SF::opt for Socket using object of type net::SF::sockOpt.
+	* 
+	* @param {net::SF::opt} _opType Option to set for Socket.
+	* @param {net::SF::sockOpt} _opValue ___________			!!Improve This!!.
```
Comment (agauniyal) [inline include/socket.hpp]: ___________ = socket option structure. Present inside socket_family.hpp

**Review by agauniyal — CHANGES_REQUESTED** [review]

**Comment by tkhurana96** [comment]
The commit msg was actually: 	Add clang formatted doc and remove all !!Improve This!! from doc

**Comment by agauniyal** [comment]
@tkhurana96 Since no one else is working on your branch, execute this - 
`git commit --amend -m "Add clang formatted doc and remove all !!Improve This!! comments"` 

and then execute -
`git push -f`.

**Comment by agauniyal** [comment]
@tkhurana96 there is `net::methods` inside `socket_family.hpp`, would you mind adding theirs too.

**Review by agauniyal — APPROVED** [review]

**Pull request merged**